litr/src/main/java/com/linkedin/android/litr/io/MediaMuxerMediaTarget.java

```diff
@@ -40,29 +44,66 @@
 
     private MediaFormat[] mediaFormatsToAdd;
 
+    private ParcelFileDescriptor parcelFileDescriptor;
     private String outputFilePath;
     private int numberOfTracksToAdd;
     private int trackCount;
 
-    public MediaMuxerMediaTarget(@NonNull String outputFilePath, @IntRange(from = 1) int trackCount, int orientationHint, int outputFormat) throws MediaTargetException {
-        this.outputFilePath = outputFilePath;
-        this.trackCount = trackCount;
-
+    /**
+     * Create an instance using input URI. On Android Oreo (API level 26) and above it can be any URI writeable by the
+     * provided context, otherwise it must be a "file://" URI. This constructor is especially useful when working with
+     * FileProvider created URI to comply with scoped storage enforcement on Android 10+
+     */
+    public MediaMuxerMediaTarget(@NonNull Context context, @NonNull Uri outputFileUri,
+            @IntRange(from = 1) int trackCount, int orientationHint, int outputFormat) throws MediaTargetException {
         try {
-            mediaMuxer = new MediaMuxer(outputFilePath,  outputFormat);
-            mediaMuxer.setOrientationHint(orientationHint);
+            MediaMuxer mediaMuxer;
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                parcelFileDescriptor = context.getContentResolver().openFileDescriptor(outputFileUri, "rwt");
+                if (parcelFileDescriptor != null) {
+                    mediaMuxer = new MediaMuxer(parcelFileDescriptor.getFileDescriptor(), outputFormat);
+                } else {
+                    throw new IOException();
```

Comment:
Maybe MediaTargetException here as well? Use IO_FAILURE enum value

Reply:
@izzytwosheds, already the `MediaTargetException` with `IO_FAILURE` will be thrown after throwing the `IOException` because there is an `IOException` catch case with the try, else if you prefer to directly throw the `MediaTargetException` here.

Reply:
Gotcha, I see it now. Should we add a helpful message to IOException? Something like "Inaccessible URI" + outputFileUri ?

Reply:
I think yes, and it's fixed now